lab6/react-spa/src/components/forecast/c_forecast.jsx

```diff
@@ -0,0 +1,378 @@
+import React, {useEffect} from 'react';
+import sun from '../all/img/sunny.jpg'
+import {useAuth} from "../all/api/useAuth";
+import {draw} from "./rose_draw";
+import Load from "../loading/loading";
+const Forecast=()=> {
+    const { isSignedIn} = useAuth()
+    const svgns = "http://www.w3.org/2000/svg";
+    (function () {
+        if (typeof window.CustomEvent === "function") return false;
+        function CustomEvent(event, params) {
+            params = params || { bubbles: false, cancelable: false, detail: null };
+            let evt = document.createEvent('CustomEvent');
+            evt.initCustomEvent(event, params.bubbles, params.cancelable, params.detail);
+            return evt;
+        }
+        window.CustomEvent = CustomEvent;
+    })();
+    const $modal = function (options) {
+        try{
+            let test = document.getElementById('fuck')
+            if(test === null ){
+                let
+                    _elemModal,
+                    _eventShowModal,
+                    _eventHideModal,
+                    _hiding = false,
+                    _destroyed = false,
+                    _animationSpeed = 200;
+
+                function _createModal(options) {
+                    let
+                        elemModal = document.createElement('div'),
+                        modalTemplate = '<div class="modal__backdrop" data-dismiss="modal"><div class="modal__content"><div class="modal__header"><div class="modal__title" data-modal="title">{{title}}</div><span class="modal__btn-close" data-dismiss="modal" title="Закрыть">×</span></div><div class="modal__body" data-modal="content">{{content}}</div>{{footer}}</div></div>',
+                        modalFooterTemplate = '<div class="modal__footer">{{buttons}}</div>',
+                        modalButtonTemplate = '<button type="button" class="{{button_class}}" data-handler={{button_handler}}>{{button_text}}</button>',
+                        modalHTML,
+                        modalFooterHTML = '';
+
+                    elemModal.classList.add('modal');
+                    elemModal.id = "fuck";
+                    let rose = '<svg xmlns="http://www.w3.org/2000/svg" width="600" height="600" viewBox="0 0 600 600"><rect width="100%" height="100%" fill="#fff" /></svg>';//document.getElementById("fake").innerHTML
+                    modalHTML = modalTemplate.replace('{{title}}', options.title || 'Simple Wind rose, according to latest data');
+                    modalHTML = modalHTML.replace('{{content}}', options.content || rose);
```

Comment:
u need to make a component called `Dialog` and then make a conditional render

Reply:
created component c_dialog